container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/JerseyServletProvider.java

```
@@ -0,0 +1,119 @@
+import java.util.Optional;
+
+import static com.yahoo.container.servlet.jersey.util.ResourceConfigUtil.registerComponent;
+
+/**
+ * @author Tony Vaagenes
+ * @author ollivir
+ */
+public class JerseyServletProvider implements Provider<ServletHolder> {
+    private final ServletHolder jerseyServletHolder;
+
+    public JerseyServletProvider(RestApiContext restApiContext) {
+        this.jerseyServletHolder = new ServletHolder(new ServletContainer(resourceConfig(restApiContext)));
+    }
+
+    private ResourceConfig resourceConfig(RestApiContext restApiContext) {
+        final ResourceConfig resourceConfig = ResourceConfig
+                .forApplication(new JerseyApplication(resourcesAndProviders(restApiContext.getBundles())));
+
+        registerComponent(resourceConfig, componentInjectorBinder(restApiContext));
+        registerComponent(resourceConfig, jacksonDatatypeJdk8Provider());
+        resourceConfig.register(MultiPartFeature.class);
+
+        return resourceConfig;
+    }
+
+    private static Collection<Class<?>> resourcesAndProviders(Collection<BundleInfo> bundles) {
+        final List<Class<?>> ret = new ArrayList<>();
+
+        for (BundleInfo bundle : bundles) {
+            for (String classEntry : bundle.getClassEntries()) {
+                Optional<String> className = detectResourceOrProvider(bundle.classLoader, classEntry);
+                className.ifPresent(cname -> ret.add(loadClass(bundle.symbolicName, bundle.classLoader, cname)));
+            }
+        }
+        return ret;
+    }
+
+    private static Optional<String> detectResourceOrProvider(ClassLoader bundleClassLoader, String classEntry) {
+        try (InputStream inputStream = getResourceAsStream(bundleClassLoader, classEntry)) {
+            ResourceOrProviderClassVisitor visitor = ResourceOrProviderClassVisitor.visit(new ClassReader(inputStream));
+            return Optional.ofNullable(visitor.getClassName());
+        } catch (IOException e) {
+            // ignored
```

Should this exception be ignored?